app/controllers/discover.js

```diff
@@ -0,0 +1,170 @@
+            });
+        });
+    },
+
+    getQueryBody() {
+        let facetFilters = this.get('facetFilters');
+        let filters = [];
+        for (let k of Object.keys(facetFilters)) {
+            let filter = facetFilters[k];
+            if (filter) {
+                if (Ember.$.isArray(filter)) {
+                    filters = filters.concat(filter);
+                } else {
+                    filters.push(filter);
+                }
+            }
+        }
+
+        let query = {
+            'query_string' : {
+                'query': this.get('searchString') || '*'
+            }
+        };
+        if (filters.length) {
+            query = {
+                'bool': {
+                    'must': query,
+                    'filter': filters
+                }
+            };
+        }
+
+        let sort = [];
+        let sortBy = this.get('sortBy') || 'Relevance';
+        if ( sortBy === 'Upload date (oldest to newest)' ) {
+            sort.push({
+                "date_updated": {"order": "asc"}
+            });
+        } else if ( sortBy === 'Upload date (newest to oldest)' ) {
+            sort.push({
+                "date_updated": {"order": "desc"}
+            });
+        }
+
+        let queryBody = {
+            query,
+            from: (this.get('page') - 1) * this.get('size'),
+            sort
+        };
+
+        return this.set('queryBody', queryBody);
+    },
+
+    togglePagination() {
+        // prev
+        if (this.get('page') > 1) {
+            this.set('showPrev', true);
+        } else {
+            this.set('showPrev', false);
+        }
+        // next
+        if (this.get('page') * this.get('size') <= this.get('numberOfResults')) {
+            this.set('showNext', true);
+        } else {
+            this.set('showNext', false);
+        }
+    },
+
+    termsFilter(field, terms, raw = true) {
+        if (terms && terms.length) {
+            if (raw) {
+                field = field + '.raw';
+            }
+            let filter = { terms: {} };
+            filter.terms[field] = terms;
+            return filter;
+        } else {
+            return null;
+        }
+    },
+
+    actions: {
+        search(query) {
+            this.set('searchString', query);
+            this.set('page', 1);
+            this.loadPage();
+        },
+
+        previous() {
+            if (this.get('page') > 1) {
+                this.decrementProperty('page');
+                this.loadPage();
+            }
+        },
+
+        next() {
+            if (this.get('page') * this.get('size') <= this.get('numberOfResults')) {
+                this.incrementProperty('page');
+                this.loadPage();
+            }
+        },
+
+        sortBy(option) {
+            this.set('sortBy', option);
+            this.set('page', 1);
+            this.loadPage();
+        },
+
+        linkToAddPreprint() {
+            this.transitionToRoute('/preprints/add');
```

Comment:
This helper should probably use view names (dot separated), not URLs (with slashes).